neuro_san/service/chat_message_converter.py

```diff
@@ -0,0 +1,49 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import Dict
+from neuro_san.internals.messages.chat_message_type import ChatMessageType
+
+
+class ChatMessageConverter:
+    """
+    Helper class to prepare chat response messages
+    for external clients consumption.
+    """
+
+    @classmethod
+    def convert(cls, response_dict: Dict[str, Any]):
+        """
+        Convert chat response message to a format expected by external clients:
+        :param response_dict: chat response message to be sent out
+        """
+        # Ensure that we return ChatMessageType as a string in output json
+        message_dict: Dict[str, Any] = response_dict.get('response', None)
+        if message_dict is not None:
+            ChatMessageConverter.convert_message(message_dict)
+
+    @classmethod
+    def convert_message(cls, message_dict: Dict[str, Any]):
+        """
+        Convert chat message to a format expected by external clients:
+        :param message_dict: chat message to process
+        """
```

Comment:
We go down ChatMessage structure recursively to handle all ChatMessageType instances.